### PR TITLE
Feature/pi 1939 mas court documents

### PR DESCRIPTION
**Pull request by achimber-moj**

### CODE REVIEW

File: projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/SentenceRepository.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package uk.gov.justice.digital.hmpps.integrations.delius.sentence.entity
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.Query
 import uk.gov.justice.digital.hmpps.integrations.delius.overview.entity.Event
+import java.time.LocalDate
```
Comment (marcus-bcl) [inline projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/SentenceRepository.kt:6]: Is this used?

Comment (marcus-bcl) [reply projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/SentenceRepository.kt:6]: If you're committing via IntelliJ, I'd recommend ticking the "Optimise imports" and "Re-format code" options in the commit dialog so it sorts these out for you automatically.

File: projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/personalDetails/entity/Document.kt [hunk 2]
```diff
@@ -59,11 +60,61 @@ class ContactDocument(
...
 interface DocumentRepository : JpaRepository<PersonDocument, Long> {
     fun findByPersonId(personId: Long): List<PersonDocument>
 
     @Query("select d.name from PersonDocument d join Person p on p.id = d.personId and p.crn = :crn and d.alfrescoId = :alfrescoId")
     fun findNameByPersonCrnAndAlfrescoId(crn: String, alfrescoId: String): String?
+
+    @Query(
+        """
+            SELECT id, lastSaved, item FROM (
+                SELECT d.DOCUMENT_ID AS id, d.LAST_SAVED AS lastSaved, NVL(DECODE(d.DOCUMENT_TYPE, 'CPS_PACK', 'CPS Pack', rdt.DESCRIPTION), 'Pre Sentence Event') AS item
```
[the remark this answers is not in the document]
Comment (achimber-moj) [reply projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/personalDetails/entity/Document.kt]: done

File: projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/SentenceRepository.kt (hunk 1 above)
Comment (achimber-moj) [reply projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/sentence/entity/SentenceRepository.kt]: done

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**